mindsdb/integrations/handlers/huggingface_handler/huggingface_handler.py

```diff
@@ -346,38 +347,143 @@ def describe(self, attribute: Optional[str] = None) -> pd.DataFrame:
             tables = ["args", "metadata"]
             return pd.DataFrame(tables, columns=["tables"])
 
-    def finetune(
-        self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None
-    ) -> None:
-        finetune_args = args if args else {}
-        args = self.base_model_storage.json_get("args")
-        args.update(finetune_args)
+    def finetune(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        logger.info("Starting finetune method")
+        self.print_debug_info()
+
+        if df is None or df.empty:
+            raise ValueError("No data provided for fine-tuning")
+
+        logger.info(f"Input DataFrame - shape: {df.shape}, columns: {df.columns}")
+
+        # Retrieve stored args, defaulting to an empty dict if None
+        stored_args = self.model_storage.json_get("args") or {}
```

Comment:
Should these stored arguments be retrieved from `self.model_storage` or `self.base_model_storage`? The original code uses `self.base_model_storage`.
CC: @dusvyat

Reply:
model_storage seems better imo